libsolidity/ast/ASTAnnotations.h

```diff
@@ -148,16 +148,21 @@ struct StructDeclarationAnnotation: TypeDeclarationAnnotation
 	std::optional<bool> containsNestedMapping;
 };
 
-struct ContractDefinitionAnnotation: TypeDeclarationAnnotation, StructurallyDocumentedAnnotation
+struct SharedContractDependenciesAnnotation
+{
+	/// List of contracts this contract or function creates, i.e. which need to be compiled first.
+	/// For ContractDefinitions: Also includes all contracts from linearizedBaseContracts.
+	/// Only set for contracts or free functions
```

Comment:
Why not internal library functions?

Reply:
it is in fact also set for library functions. I kind of counted those as contracts. I'll make the comment more explicit